### PR TITLE
Test in ChefDK and use cookstyle

### PR DESCRIPTION
**Pull request by tas50**
Update testing and introduce failing on deprecations so we fail if we would fail to run on a future chef version

### CODE REVIEW

**Comment by sneal** [comment]
`FC009: Resource attribute not recognised: ./recipes/default.rb:21` so apparently FoodCritic still doesn't like the package resource on Windows. Hopefully the latest version does and we can just bump the FC version.

File: Gemfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ gem 'rspec-expectations'
 gem 'rspec-mocks'
 gem 'chef'
 gem 'foodcritic'
-gem 'rubocop'
+gem 'cookstyle'
```
Comment (sneal) [inline Gemfile]: Cool! I haven't kept up with Chef lately, good to see tooling more specific to Chef here.

**Comment by tas50** [comment]
@sneal Let's start simpler here, by just getting testing updated a bit. This gets us the latest chefDK and testing using just ChefDK vs. the hybrid gem/chefDK that existed before.

**Comment by sneal** [comment]
:+1: Passes now, although I see you've reverted the `windows_package` change.

**Comment by tas50** [comment]
It appears as though we need attributes that still force calling out the actual underlying resource which is windows_package still.

**Comment by tas50** [comment]
@sneal can we get this merged?

File: .kitchen.appveyor.yml [hunk 2]
```diff
@@ -16,6 +16,6 @@ platforms:
       box: mwrock/Windows2012R2
```
Comment (iennae) [inline .kitchen.appveyor.yml]: Just curious, but what's special about matt's box there?

File: libraries/matchers.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #
 # Author:: Shawn Neal (<sneal@sneal.net>)
-# Cookbook Name:: visualstudio
+# Cookbook:: visualstudio
```
Comment (iennae) [inline libraries/matchers.rb]: This looks like a copy pasta comment error with regards to the cookbook name (I know it's not to do with this PR specifcally just might want to fix it as it's a bit confusing.

File: .kitchen.appveyor.yml (hunk 2 above)
Comment (sneal) [inline .kitchen.appveyor.yml]: Not much, but it's public and has winrm setup

File: libraries/matchers.rb (hunk 3 above)
Comment (sneal) [inline libraries/matchers.rb]: Thanks! 🍝

**Comment by sneal** [comment]
Thanks @tas50 lets :shipit:

**Pull request merged**